src/app.py

```diff
@@ -148,6 +150,32 @@ def watermark(self, file):
         # copio il file temporaneo nella cartella di output
         os.replace(temp, self.outputpath + "/" + os.path.basename(file))
 
+    def is_format_A4(self, file_path) -> bool:
+            # This function check wether the file is indeed in A4 format. 
+            # We have noticed that the watermark is not placed correctly if 
+            # the format is different from A4.
+
+            # A4 Format is width = 595.0 and height = 842.0
+            # These constants are defined in the constants.py file
+            # We check if the width and height of the page are
+            # within a chosen tolerance. 
+            tol = 1
+
+            with open(file_path, "rb") as file: 
+                reader = PdfReader(file)
+                page_num = len(reader.pages)
+                for idx in range(page_num):
+                    page = reader.pages[idx]
+
+                    width = page.mediabox.width
+                    height = page.mediabox.height
+
+                    if abs(width - constants.a4_width) < tol and abs(height - constants.a4_height) < tol:
+                        return True
+                    
+            return False
```

Comment:
Qui se un file ha la prima pagina A4 e la seconda non A4 restituisce vero, dovresti mettere
```
for ...:
    ...
    if ~(condizione di A4):
        return False
return True
```

Reply:
Should've fixed